### PR TITLE
Us29 cleanup ai systems

### PR DESCRIPTION
**Pull request by Heso113**
Added comments to the systems where it was missing. Fixed the bugg with the optimized FindClosestTile function.

### CODE REVIEW

File: Game/GameMain/gameAI/AISystems.cpp [hunk 1]
```diff
@@ -383,19 +385,24 @@ int2 ecs::systems::PathfindingStateSystem::GetClosestTile(TransformComponent& tr
...
 				if (i >= 0 && i < p_gp->GetSize().y && j >= 0 && j < p_gp->GetSize().x)
 				{
-					p_curr_tile_transform = ECSUser::getComponentFromKnownEntity<TransformComponent>(p_gp->mGrid[tile_index.y][tile_index.x].Id);
-					distance = PhysicsHelpers::CalculateDistance(p_curr_tile_transform->position, transform.position);
+					p_curr_tile_transform = ECSUser::getComponentFromKnownEntity<TransformComponent>(p_gp->mGrid[j][i].Id);
+					XMFLOAT3 temp_pos = p_curr_tile_transform->position;
+					temp_pos.y = 0.0f;
+					XMFLOAT3 temp_pos2 = transform.position;
+					temp_pos2.y = 0.0f;
+					distance = PhysicsHelpers::CalculateDistance(temp_pos, temp_pos2);
```
Comment (Wessebanan) [inline Game/GameMain/gameAI/AISystems.cpp]: mer beskrivande än temp_pos mayhaps?

File: Game/GameMain/gameAI/AISystems.cpp [hunk 2]
```diff
@@ -1043,27 +1052,84 @@ ID ecs::systems::MoveStateSystem::GetClosestTileId(TransformComponent& transform
...
```
Comment (Wessebanan) [inline Game/GameMain/gameAI/AISystems.cpp]: mer beskrivande än temp_pos mayhaps?

File: Game/GameMain/gameAI/AISystems.cpp [hunk 3]
```diff
@@ -1043,27 +1052,84 @@ ID ecs::systems::MoveStateSystem::GetClosestTileId(TransformComponent& transform
...
+			{
+				//Check if the tile is a tile we can walk on.
+				if (p_gp->mGrid[y][x].isPassable)
+				{
+					p_curr_tile_transform = ecs::ECSUser::getComponentFromKnownEntity<ecs::components::TransformComponent>(p_gp->mGrid[y][x].Id);
+					XMFLOAT3 temp_pos = p_curr_tile_transform->position;
+					temp_pos.y = 0.0f;
+					XMFLOAT3 temp_pos2 = transform.position;
+					temp_pos2.y = 0.0f;
+					temp_dist = PhysicsHelpers::CalculateDistance(temp_pos, temp_pos2);
```
Comment (Wessebanan) [inline Game/GameMain/gameAI/AISystems.cpp]: T E M P

File: Game/GameMain/gameAI/AISystems.cpp (hunk 1 above)
Comment (Heso113) [reply Game/GameMain/gameAI/AISystems.cpp]: no

File: Game/GameMain/gameAI/AISystems.cpp (hunk 2 above)
Comment (Heso113) [reply Game/GameMain/gameAI/AISystems.cpp]: no

File: Game/GameMain/gameAI/AISystems.cpp (hunk 3 above)
Comment (Heso113) [reply Game/GameMain/gameAI/AISystems.cpp]: no

**Pull request merged**